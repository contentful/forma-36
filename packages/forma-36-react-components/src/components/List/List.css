.List {
  margin: 0;
}

.List--unordered {
  list-style-type: disc;
}
