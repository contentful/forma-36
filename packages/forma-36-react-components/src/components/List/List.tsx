import React from 'react';
import cn from 'classnames';

import styles from './List.css';

type ListType = 'ul' | 'ol';

export interface ListProps {
  element?: ListType;
  className?: string;
  children: React.ReactNode;
  style?: React.CSSProperties;
  testId?: string;
}

export function List({
  element: Tag = 'ul',
  className,
  children,
  testId = 'cf-ui-list',
  ...otherProps
}: ListProps): React.ReactElement {
  const classNames = cn(
    styles['List'],
    { [styles['List--unordered']]: Tag === 'ul' },
    className,
  );

  return (
    <Tag {...otherProps} className={classNames} data-test-id={testId}>
      {children}
    </Tag>
  );
}

export default List;
