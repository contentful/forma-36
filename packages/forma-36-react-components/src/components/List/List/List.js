import React from 'react';
import cn from 'classnames';
import PropTypes from 'prop-types';
import styles from './List.css';

export class List extends React.Component {
  static propTypes = {
    extraClassNames: PropTypes.string,
    children: PropTypes.node.isRequired,
    testId: PropTypes.string,
  };

  static defaultProps = {
    extraClassNames: undefined,
    testId: 'cf-ui-list',
  };

  render() {
    const { extraClassNames, children, testId, ...otherProps } = this.props;

    const classNames = cn(styles.List, extraClassNames);

    return (
      <ul className={classNames} data-test-id={testId} {...otherProps}>
        {children}
      </ul>
    );
  }
}

export default List;
