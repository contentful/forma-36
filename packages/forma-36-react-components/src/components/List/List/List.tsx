import React, { Component } from 'react';
import cn from 'classnames';
import styles from './List.css';

export interface ListProps {
  extraClassNames?: string;
  children: React.ReactNode;
  testId?: string;
}

export class List extends Component<ListProps> {
  static defaultProps = {
    testId: 'cf-ui-list',
  };

  render() {
    const { extraClassNames, children, testId, ...otherProps } = this.props;

    const classNames = cn(styles['List'], extraClassNames);

    return (
      <ul className={classNames} data-test-id={testId} {...otherProps}>
        {children}
      </ul>
    );
  }
}

export default List;
