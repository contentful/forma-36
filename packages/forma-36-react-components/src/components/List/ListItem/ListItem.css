@import 'settings/colors';
@import 'settings/typography';
@import 'settings/transitions';

.ListItem {
  font-size: var(--font-size-m);
  font-family: var(--font-stack-primary);
  line-height: var(--line-height-default);
}

.ListItem--nested-list {
  list-style: none;
}
