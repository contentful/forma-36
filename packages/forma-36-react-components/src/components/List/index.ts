export * from './List';
export { default } from './List';
