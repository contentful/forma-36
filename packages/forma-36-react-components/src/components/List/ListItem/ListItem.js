import React from 'react';
import cn from 'classnames';
import PropTypes from 'prop-types';
import styles from './ListItem.css';
import List from '../List/List';

export class ListItem extends React.Component {
  static propTypes = {
    extraClassNames: PropTypes.string,
    children: PropTypes.node.isRequired,
    testId: PropTypes.string,
  };

  static defaultProps = {
    extraClassNames: undefined,
    testId: 'cf-ui-list-item',
  };

  render() {
    const { extraClassNames, children, testId, ...otherProps } = this.props;

    const classNames = cn(styles.ListItem, extraClassNames, {
      [styles['ListItem--nested-list']]: this.props.children.type === List,
    });

    return (
      <li className={classNames} data-test-id={testId} {...otherProps}>
        {children}
      </li>
    );
  }
}

export default ListItem;
