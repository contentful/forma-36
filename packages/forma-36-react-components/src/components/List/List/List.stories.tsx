import React from 'react';
import { storiesOf } from '@storybook/react';
import { text } from '@storybook/addon-knobs/react';

import List from './List';
import ListItem from '../ListItem';
import TextLink from '../../TextLink';

storiesOf('Components|List', module)
  .addParameters({
    propTypes: [List['__docgenInfo'], ListItem['__docgenInfo']],
  })
  .add('default', () => (
    <List extraClassNames={text('Extra Class Names', '')}>
      <ListItem>List Item 1</ListItem>
      <ListItem>List Item 2</ListItem>
      <ListItem>
        List Item with a&nbsp;
        <TextLink>text link</TextLink>&nbsp;
      </ListItem>
      <ListItem>
        <List>
          <ListItem>Sublist Item 1</ListItem>
          <ListItem>Sublist Item 2</ListItem>
        </List>
      </ListItem>
    </List>
  ));
