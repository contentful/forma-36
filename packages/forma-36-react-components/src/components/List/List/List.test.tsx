import React from 'react';
import { shallow, mount } from 'enzyme';
import { axe } from 'jest-axe';
import List from './List';
import ListItem from '../ListItem';

it('renders the component', () => {
  const output = shallow(
    <List>
      <ListItem>Item</ListItem>
    </List>,
  );

  expect(output).toMatchSnapshot();
});

it('renders the component with an additional class name', () => {
  const output = shallow(
    <List extraClassNames="my-extra-class">
      <ListItem>Item</ListItem>
    </List>,
  );

  expect(output).toMatchSnapshot();
});

it('has no a11y issues', async () => {
  const output = mount(
    <List>
      <ListItem>Item</ListItem>
    </List>,
  ).html();
  const results = await axe(output);

  expect(results).toHaveNoViolations();
});
