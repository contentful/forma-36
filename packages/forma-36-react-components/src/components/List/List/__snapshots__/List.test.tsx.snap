// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the component 1`] = `
<ul
  className="List"
  data-test-id="cf-ui-list"
>
  <ListItem
    testId="cf-ui-list-item"
  >
    Item
  </ListItem>
</ul>
`;

exports[`renders the component with an additional class name 1`] = `
<ul
  className="List my-extra-class"
  data-test-id="cf-ui-list"
>
  <ListItem
    testId="cf-ui-list-item"
  >
    Item
  </ListItem>
</ul>
`;
